cognite/client/data_classes/sequences.py

```diff
@@ -145,3 +179,136 @@ def remove(self, value: List) -> SequenceUpdate:
 class SequenceList(CogniteResourceList):
     _RESOURCE = Sequence
     _UPDATE = SequenceUpdate
+
+
+class SequenceData:
+    """An object representing a list of rows from a sequence.
+
+    Args:
+        id (int): Id of the sequence the data belong to
+        external_id (str): External id of the sequence the data belong to
+        rows (List[dict]): combined row numbers and row data object from the API. If you pass this, row_numbers/values are ignored.
+        row_numbers (List[int]): The data row numbers.
+        values (List[List[ Union[int, str, float]]]): The data values, one row at a time.
+        columns: List[dict]: The column information, in the format returned by the API
+    """
+
+    def __init__(
+        self,
+        id: int = None,
+        external_id: str = None,
+        rows: List[dict] = None,
+        row_numbers: List[int] = None,
+        values: List[List[Union[int, str, float]]] = None,
+        columns: List[dict] = None,
+    ):
+        if rows:
+            row_numbers = [r["rowNumber"] for r in rows]
+            values = [r["values"] for r in rows]
+        self.id = id
+        self.external_id = external_id
+        self.row_numbers = row_numbers or []
+        self.values = values or []
+        self.columns = columns
+
+    def __str__(self):
+        return json.dumps(self.dump(), indent=4)
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __len__(self) -> int:
+        return len(self.row_numbers)
+
+    def __eq__(self, other):
+        return (
+            type(self) == type(other)
+            and self.id == other.id
+            and self.external_id == other.external_id
+            and self.row_numbers == other.row_numbers
+            and self.values == other.values
+        )
+
+    def __getitem__(self, item) -> List[Union[int, str, float]]:
+        # slow, should be replaced by dict cache if it sees more than incidental use
+        return self.values[self.row_numbers.index(item)]
+
+    def __getattr__(self, item) -> List[Union[int, str, float]]:
+        """Get a column by externalId
+        """
+        try:
+            ix = self.column_external_ids().index(item)
+        except ValueError as e:
+            raise ValueError(
+                "Column {} not found, Sequence column external ids are {}".format(item, self.column_external_ids())
+            )
+        return [r[ix] for r in self.values]
+
+    def __iter__(self) -> Generator[Tuple[int, List[Union[int, str, float]]], None, None]:
+        for row, values in zip(self.row_numbers, self.values):
+            yield row, values
```

Comment:
Feel free to disagree here, I'm not very adamant about this. But I feel that this should have the same return type as `__getitem__` per yielded item, so not return the row number. But it may be useful, I don't know.

Reply:
I think you give getitem a row number already, so getting it back is strange. But iterating over pairs is natural for this data type, since it's not a list.

Reply:
But you are implementing all the list magic methods, so it should behave like a list.

Reply:
it's kind of behaving like a dict. we could call this iteritems instead if that makes you happy

Reply:
.items() in that case, which is the method you have on a dict.

Reply:
dict has items (returns the entire object) and iteritems (returns an iterator), so it's the latter I think

Reply:
You are correct.